gobblin-data-management/src/main/java/gobblin/data/management/retention/dataset/ConfigurableCleanableDataset.java

```diff
@@ -48,39 +50,58 @@
   private final VersionFinder<? extends T> versionFinder;
   private final RetentionPolicy<T> retentionPolicy;
 
+  /**
+   * Creates a new ConfigurableCleanableDataset configured through gobblin-config-management. The constructor expects
+   * {@link #VERSION_FINDER_CLASS_KEY} and {@link #RETENTION_POLICY_CLASS_KEY} to be available in the
+   * <code>config</code> passed.
+   */
+  public ConfigurableCleanableDataset(FileSystem fs, Properties jobProps, Path datasetRoot, Config config)
```

Comment:
Added Logger as an argument.
Yes the two constructors are similar but PreConditions check is different. One does config.hasPath and other calls props.containsKey. I am not sure how we can make it modular.